File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/chain_of_responsibility/lab/CustomerRequestsDiscountEvent.java
// CustomerRequestsDiscountEvent.java

public class CustomerRequestsDiscountEvent
{
	public CustomerRequestsDiscountEvent(int aPercent)
	{
		setPercent(aPercent);
	}

	private void setPercent(int aPercent)
	{
		myPercent = aPercent;
	}

	public int getPercent()
	{
		return myPercent;
	}

	private int myPercent;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/chain_of_responsibility/lab/customer_requests_discount_event.cpp
// customer_requests_discount_event.cpp

#include "customer_requests_discount_event.h"

customer_requests_discount_event::customer_requests_discount_event(int a_percent)
{
	set_percent(a_percent);
}

void customer_requests_discount_event::set_percent(int a_percent)
{
	my_percent = a_percent;
}

int customer_requests_discount_event::get_percent() const
{
	return my_percent;
}
